pyethapp/tests/test_jsonrpc.py

```diff
@@ -205,7 +209,12 @@ def rpc_request(self, method, *args):
     def fin():
         log.debug('stopping test app')
         for service in app.services:
-            app.services[service].stop()
+            gevent.sleep(.1)
+            try:
+                app.services[service].stop()
+            except Exception as e:
+                log.DEV(str(e), exc_info=e)
+                pass
```

Comment:
I hadn't had any problems anymore with above finalizer. But it also probably won't hurt to clean up thoroughly.